function savetodb(data){
    return new Promise((resolve, reject)=>{
    let internetspeed=Math.floor(Math.random()*10)+1;
    if(internetspeed>4){
        resolve("data was saved");
    }else{
        reject("failure, weak connection");
    }
    });
}
savetodb("rimsha")
.then((result)=>{
console.log("Data1 saved :promise was resolved");
console.log("result of promise:",result);
return savetodb("anna");
})
.then((result)=>{
console.log("Data2 saved :promise was resolved");
console.log("result of promise:",result);
return savetodb("bushra");
})
.then((result)=>{
console.log("Data3 saved :promise was resolved");
console.log("result of promise:",result);
})
.catch((error)=>{
    console.log("failure:promise was rejected");
    console.log("result of failure:",error);
})
